operators/src/upgrade/controller/reconciler.rs

```diff
@@ -216,15 +221,32 @@ impl ResourceContext {
             .await?;
 
         let pods: Api<Pod> = Api::namespaced(
-            UpgradeOperatorConfig::get_config().k8s_client().client(),
-            UpgradeOperatorConfig::get_config().namespace(),
+            UpgradeConfig::get_config().k8s_client().client(),
+            UpgradeConfig::get_config().namespace(),
         );
+
+        let io_engine_listparam = ListParams::default().labels(IO_ENGINE_POD_LABEL);
+        let initial_io_engine_pod_list: ObjectList<Pod> = pods.list(&io_engine_listparam).await?;
+
+        let mut initial_io_engine_pod_uids: HashSet<&String> =
+            HashSet::with_capacity(initial_io_engine_pod_list.iter().count());
+        for pod in initial_io_engine_pod_list.iter() {
+            match pod.metadata.uid.as_ref() {
+                Some(uid) => {
+                    // Building a set of io-engine Pod uid values. These values may be used to check
+                    // for io-engine Pod restarts.
+                    initial_io_engine_pod_uids.insert(uid);
+                }
+                None => {
+                    // It's ok to panic here because all kubernetes objects should have uids.
+                    panic!("Could not list all io-engine Pods' metadata.uids before restart");
```

Comment:
Nit: it's fine but since this functions has Result<_,Error> we could still return an Error though?

Reply:
Yes, okay, done!